MathLib/LinAlg/Eigen/EigenLinearSolver.cpp

```diff
@@ -115,25 +162,16 @@ EigenLinearSolver::EigenLinearSolver(
 
     // TODO for my taste it is much too unobvious that the default solver type
     //      currently is SparseLU.
-    switch (_option.solver_type)
-    {
-    case EigenOption::SolverType::SparseLU: {
-        using SolverType = Eigen::SparseLU<Matrix, Eigen::COLAMDOrdering<int>>;
-        _solver.reset(new details::EigenDirectLinearSolver<SolverType>);
-        break;
-    }
-    case EigenOption::SolverType::BiCGSTAB: {
-        using SolverType = Eigen::BiCGSTAB<Matrix, Eigen::DiagonalPreconditioner<double>>;
-        _solver.reset(new details::EigenIterativeLinearSolver<SolverType>);
-        break;
-    }
-    case EigenOption::SolverType::CG: {
-        using SolverType = Eigen::ConjugateGradient<Matrix, Eigen::Lower, Eigen::DiagonalPreconditioner<double>>;
-        _solver.reset(new details::EigenIterativeLinearSolver<SolverType>);
-        break;
-    }
-    case EigenOption::SolverType::INVALID:
-        OGS_FATAL("Invalid Eigen linear solver type. Aborting.");
+    switch (_option.solver_type) {
```

Comment:
Switch with two cases is an if-else...
Is the todo comment above still correct? I mean there is (non-obvious) OGS_FATAL in createIterativeSolver...